### PR TITLE
MongoDB Atlas - GCP VPC Peering example

### PR DESCRIPTION
**Pull request by nikhil-mongo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: examples/MongoDB Atlas-GCP VPC Peering/Readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Example - GCP and MongoDB Atlas VPC Peering
+
+This project aims to provide an example of using Azure and MongoDB Atlas together.
```
Comment (themantissa) [inline examples/MongoDB Atlas-GCP VPC Peering/Readme.md]: Bad copy/paste (azure/gcp)

File: examples/MongoDB Atlas-GCP VPC Peering/atlas-network.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  name = "default"
+}
+
+# Create the GCP peer
+resource "google_compute_network_peering" "peering" {
+  name         = "peering-gcp-terraform-test"
+  network      = data.google_compute_network.default.self_link
+  peer_network = "https://www.googleapis.com/compute/v1/projects/${mongodbatlas_network_peering.test.atlas_gcp_project_id}/global/networks/${mongodbatlas_network_peering.test.atlas_vpc_name}"
+}
+resource "mongodbatlas_project_ip_access_list" "test" {
```
Comment (themantissa) [inline examples/MongoDB Atlas-GCP VPC Peering/atlas-network.tf:34]: Would be nice to have an explaining comment - I  know the comment in the resource explains but just for readability would likely help.

File: examples/MongoDB Atlas-GCP VPC Peering/cluster.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
+
+resource "mongodbatlas_cluster" "cluster-test-single" {
```
Comment (themantissa) [inline examples/MongoDB Atlas-GCP VPC Peering/cluster.tf]: May want to provide a comment around what this cluster example shows too.  I like how complex it is but a new user could likely use a general description.

**Review by themantissa — CHANGES_REQUESTED** [review]
Looks really good!  Just one change that's really blocking and that's the Azure mention :)  Once that's fixed can approve.  Thank you!

File: examples/MongoDB Atlas-GCP VPC Peering/Readme.md (hunk 1 above)
Comment (nikhil-mongo) [reply examples/MongoDB Atlas-GCP VPC Peering/Readme.md]: typo fixed.

File: examples/MongoDB Atlas-GCP VPC Peering/atlas-network.tf (hunk 2 above)
Comment (nikhil-mongo) [reply examples/MongoDB Atlas-GCP VPC Peering/atlas-network.tf:34]: added reference to `self_link`, googleapi URL and `ip_access_list`

File: examples/MongoDB Atlas-GCP VPC Peering/cluster.tf (hunk 3 above)
Comment (nikhil-mongo) [reply examples/MongoDB Atlas-GCP VPC Peering/cluster.tf]: add a single line comment for what this cluster is.

**Review by themantissa — APPROVED** [review]
LGTM!

**Pull request merged**